pandas/tests/series/methods/test_sort_index.py

```diff
@@ -135,3 +135,35 @@ def test_sort_index_intervals(self):
             [3, 2, 1, np.nan], IntervalIndex.from_arrays([3, 2, 1, 0], [4, 3, 2, 1])
         )
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "original_list, sorted_list, ascending, ignore_index, output_index",
+        [
+            ([2, 3, 6, 1], [2, 3, 6, 1], True, True, [0, 1, 2, 3]),
+            ([2, 3, 6, 1], [2, 3, 6, 1], True, False, [0, 1, 2, 3]),
+            ([2, 3, 6, 1], [1, 6, 3, 2], False, True, [0, 1, 2, 3]),
+            ([2, 3, 6, 1], [1, 6, 3, 2], False, False, [3, 2, 1, 0]),
+        ],
+    )
+    def test_sort_index_ignore_index(
+        self, original_list, sorted_list, ascending, ignore_index, output_index
+    ):
+        # GH 30114
+        # GH 30114
+        sr = Series(original_list)
+        expected = Series(sorted_list, index=output_index)
+
+        # Test when inplace is False
```

Comment:
can inplace be parameterised easily

Reply:
emm, i think in this case, if using parameterization, might not look very clean

Reply:
I think this should be parametrized - what do you think makes it less readable?

Reply:
ok with trying parameterize as a followup, as for .sort_values and .duplicated these tests look very much like this.

Reply:
emm, okay! i thought with `inplace` being `Ture`, we do not need to do something like `df = df.sort_index(inplace=True)`, but `False` needs to be like this, and testing object will be different, one is testing if original changes, another is to test if assigned new df is changed. 

Sorry to bring up discussion here, it definitely can be parametrized, was just my preference on such case. I will parametrize all three methods in the follow-up PR for sure!